src/wled2.ts

```diff
@@ -48,13 +48,97 @@ export = (RED: Red): void => {
     });
   });
 
+  function which(val1: any, val2: any, wrapper?: any): any {

```

Comment:
This is clever and definitely helps abstract away the messiness of my old code. The names of everything should get updated so it's very clear what they are for.

Even though this is technically a generic method that could be used in any app, in this app the first parameter will always be the payload value and the second will always be the setting value. Make that clear in the parameter names instead of `val1` and `val2` so someone reading the function will have an easier time with the big conditional stuff later on.

Same with the function name: `which` could mean anything. Function names should clearly describe what the method does.